### PR TITLE
Enable ppc64le in openresty

### PR DESCRIPTION
**Pull request by krishvoor**
I hereby granted the copyright of the changes in this pull request
to the authors of this Openresty project.
#####

This PR fixes issue #731

For every PR, an equivalent Travis job should be triggered for ppc64le architecture.
Currently, this PR only enables build job, we are working towards fixing some of
the random test failures observed on ppc64le.

Signed-off-by: Krishna Harsha Voora <krishvoor@in.ibm.com>

I hereby granted the copyright of the changes in this pull request
to the authors of this openresty project.

### CODE REVIEW

**Comment by krishvoor** [comment]
@xiaocang - Please help review.

**Comment by xiaocang** [comment]
It looks like some unit tests did not pass, would you like to fix it?

**Comment by krishvoor** [comment]
Hi, @xiaocang Thanks for your review -
Currently, the issue is not related to this PR, however, we will attempt to fix this issue separately.

**Comment by xiaocang** [comment]
@krishvoor It looks like pcre jit still doesn't support on ppc64le architecture, would you like to fix that?

```
pcre_jit_compile.c:65:2: error: #error Unsupported architecture
 #error Unsupported architecture
```

**Comment by krishvoor** [comment]
@xiaocang Can you point me to the location/URL, where this failure is observed?

**Comment by agentzh** [comment]
@krishvoor if tests are failing in CI, then we cannot enable it since it'll break CI for everyone.

**Comment by krishvoor** [comment]
@xiaocang @agentzh 
To clarify further, the reported are arising out of s390x builds,
https://travis-ci.org/github/openresty/openresty/jobs/773127763
https://travis-ci.org/github/openresty/openresty/jobs/773127764
https://travis-ci.org/github/openresty/openresty/jobs/773127765

whereas on ppc64le, the builds are happening successfully -

https://travis-ci.org/github/openresty/openresty/jobs/773127766
https://travis-ci.org/github/openresty/openresty/jobs/773127767
https://travis-ci.org/github/openresty/openresty/jobs/773127768

I will bring this to the notice of the required people from Z Team, ASAP.

**Comment by xiaocang** [comment]
Hi @krishvoor , We have recently migrated our ci to travis-ci.com, if a new commit is triggered in PR, this update should take effect.
I manually reran this ci and the result is here: https://travis-ci.com/github/openresty/openresty/builds/229665517
I checked the current master branch, the ci of the s390x architecture is also passed, would you mind rebase the latest master branch and retrigger the ci?
thanks a lot

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,24 @@ linux-s390x: &linux-s390x
             - libpcre3-dev
             - mercurial
             - libpq-dev
```
Comment (xiaocang) [inline .travis.yml:36]:
```suggestion
            - libpq-dev
            
              install:
                - sudo cpanm --notest Test::Nginx IPC::Run3 > build.log 2>&1 || (cat build.log && exit 1)
                - if [ ! -f download-cache/openssl-$OPENSSL_VER.tar.gz ]; then wget -P download-cache https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz || wget -P download-cache https://www.openssl.org/source/old/${OPENSSL_VER//[a-z]/}/openssl-$OPENSSL_VER.tar.gz; fi
                - tar zxf download-cache/openssl-$OPENSSL_VER.tar.gz
                - cd openssl-$OPENSSL_VER/
                - patch -p1 < ../patches/openssl-$OPENSSL_PATCH_VER-sess_set_get_cb_yield.patch
                - ./config no-threads shared enable-ssl3 enable-ssl3-method -g --prefix=$OPENSSL_PREFIX -DPURIFY > build.log 2>&1 || (cat build.log && exit 1)
                - make -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
                - sudo make PATH=$PATH install_sw > build.log 2>&1 || (cat build.log && exit 1)
                - cd ..
                - sudo ln -s /usr/bin/make /usr/bin/gmake
              script:
                - util/mirror-tarballs > build.log 2>&1 || (cat build.log && exit 1)
                - cd openresty-*
                - ./configure --prefix=$OPENRESTY_PREFIX --with-cc-opt="-I$PCRE_INC -I$OPENSSL_INC" --with-ld-opt="-L$PCRE_LIB -L$OPENSSL_LIB -Wl,-rpath,$PCRE_LIB:$OPENSSL_LIB" --with-pcre-jit --with-http_ssl_module --with-debug -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
                - make -j$JOBS > build.log 2>&1 || (cat build.log && exit 1)
                - sudo make install > build.log 2>&1 || (cat build.log && exit 1)
                - cd ..
                - export PATH=$OPENRESTY_PREFIX/bin:$OPENRESTY_PREFIX/nginx/sbin:$PATH
                - nginx -V
                - ldd `which nginx`|grep -E 'luajit|ssl|pcre'
```

**Comment by krishvoor** [comment]
Hi @xiaocang
Thank you for your suggestion -

I have rebased on top of the master and the jobs got triggered, here's the summary -
https://travis-ci.com/github/openresty/openresty/builds/231457948.

which shows the failure of s390x jobs, however, 
the changes from my PR only enables ppc64le jobs to be triggered.
can you kindly tell me in case I have overlooked and is breaking the jobs for s390x today?

**Comment by xiaocang** [comment]
Hi @krishvoor, would you mind using my patch to fix it so that the CI should be able to pass

**Comment by krishvoor** [comment]
@xiaocang Thanks for your comments, will apply the changes.

**Comment by krishvoor** [comment]
Hi, @xiaocang Apologies for getting back late onto this.
I applied your changes, and are now in. The changes are rebased against the master branch as well.
Please help trigger/ invoke the Travis-CI jobs, with this last set of changes the CI should work for all.

File: .travis.yml [hunk 2]
```diff
@@ -34,6 +34,28 @@ linux-s390x: &linux-s390x
...
+            
+              install:
```
Comment (xiaocang) [inline .travis.yml:38]: `install:` block should be consistent with the indentation of `addons:`

**Review by xiaocang — CHANGES_REQUESTED** [review]

**Comment by xiaocang** [comment]
Hi @krishvoor, thanks for the contribution, there is still some yml formatting to change.

File: .travis.yml (hunk 2 above)
Comment (krishvoor) [reply .travis.yml]: Missed this, thanks for the suggestion.
The indentation is in line with `addons:`

**Review by doujiang24 — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -84,6 +124,12 @@ jobs:
           env: OPENSSL_VER=1.1.0l OPENSSL_PATCH_VER=1.1.0d
         - <<: *linux-s390x
           env: OPENSSL_VER=1.1.1k OPENSSL_PATCH_VER=1.1.1f
+        - <<: *linux-ppc64le
```
Comment (zhuizhuhaomeng) [inline .travis.yml:127]: I don‘t think there is a need to test openssl 1.0.2 and 1.1.0 anymore.

**Review by zhuizhuhaomeng — APPROVED** [review]

File: .travis.yml (hunk 3 above)
Comment (krishvoor) [reply .travis.yml:127]: Ok, so the below SSL version jobs are no longer needed?

```
    - OPENSSL_VER=1.0.2u OPENSSL_PATCH_VER=1.0.2h
    - OPENSSL_VER=1.1.0l OPENSSL_PATCH_VER=1.1.0d
```
I can remove the corresponding jobs if not needed.

**Comment by xiaocang** [comment]
Merged in c7a3cfe with the following patch:
Thanks for your contribution!

``` patch
diff --git a/.travis.yml b/.travis.yml
index 9ab9ac5..b55640a 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -34,7 +34,6 @@ linux-s390x: &linux-s390x
             - libpcre3-dev
             - mercurial
             - libpq-dev
-            
   install:
     - sudo cpanm --notest Test::Nginx IPC::Run3 > build.log 2>&1 || (cat build.log && exit 1)
     - if [ ! -f download-cache/openssl-$OPENSSL_VER.tar.gz ]; then wget -P download-cache https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz || wget -P download-cache https://www.openssl.org/source/old/${OPENSSL_VER//[a-z]/}/openssl-$OPENSSL_VER.tar.gz; fi
@@ -124,10 +123,6 @@ jobs:
           env: OPENSSL_VER=1.1.0l OPENSSL_PATCH_VER=1.1.0d
         - <<: *linux-s390x
           env: OPENSSL_VER=1.1.1k OPENSSL_PATCH_VER=1.1.1f
-        - <<: *linux-ppc64le
-          env: OPENSSL_VER=1.0.2u OPENSSL_PATCH_VER=1.0.2h
-        - <<: *linux-ppc64le
-          env: OPENSSL_VER=1.1.0l OPENSSL_PATCH_VER=1.1.0d
         - <<: *linux-ppc64le
           env: OPENSSL_VER=1.1.1k OPENSSL_PATCH_VER=1.1.1f
```

**Pull request closed without merging**